### ISSUE TITLE
Upgrade to 1.9.4.3 breaks grids using `filter_condition_callback` SUPEE-11219 [confirmed+fix]

### ISSUE THREAD

**Issue by seansan** [OpenMage/magento-lts#882]
Upgrade to 1.9.4.3 breaks grids using `filter_condition_callback` SUPEE-11219 [confirmed]

Due to code changes here https://github.com/OpenMage/magento-lts/commit/cce46fced84e2e61d20fa7227b80f1f652df538a#r36881190 belonging to SUPEE-11219

New problems occur in various extensions a code utilizing `filter_condition_callback` as described here and as ... well we have always done

To me this seems as a breaking change ... and as such breaking more than necessary .. although I must say it is not clear why the extra if check was added in the first place

The problem is described here: https://magento.stackexchange.com/a/292307/78 and here in more (user) detail https://magento.stackexchange.com/questions/294824/magento-1-9-4-3-custom-column-filter-not-working

Expected
Custom filters to work in grids

Actual
Custom filters do no longer work

Solution
Remove the second if statement again

![image](https://user-images.githubusercontent.com/652395/72530712-3bdc3000-3870-11ea-9b49-1fbebef52673.png)

The error is 

```
invalid attribute name: warehouse_ids

#0 app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php(1295): Mage::exception('Mage_Eav', 'Invalid attribu...')
#1 app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php(1388): Mage_Eav_Model_Entity_Collection_Abstract->_addAttributeJoin('warehouse_ids', 'inner')
#2 app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php(322): Mage_Eav_Model_Entity_Collection_Abstract->_getAttributeConditionSql('warehouse_ids', Array, 'inner')
#3 app/code/core/Mage/Catalog/Model/Resource/Product/Collection.php(1446): Mage_Eav_Model_Entity_Collection_Abstract->addAttributeToFilter('warehouse_ids', Array, 'inner')
#4 app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php(342): Mage_Catalog_Model_Resource_Product_Collection->addAttributeToFilter('warehouse_ids', Array)
#5 app/code/core/Mage/Adminhtml/Block/Widget/Grid.php(472): Mage_Eav_Model_Entity_Collection_Abstract->addFieldToFilter('warehouse_ids', Array)
#6 app/code/core/Mage/Adminhtml/Block/Catalog/Product/Grid.php(142): Mage_Adminhtml_Block_Widget_Grid->_addColumnFilterToCollection(Object(Mage_Adminhtml_Block_Widget_Grid_Column))
#7 app/code/core/Mage/Adminhtml/Block/Widget/Grid.php(457): Mage_Adminhtml_Block_Catalog_Product_Grid->_addColumnFilterToCollection(Object(Mage_Adminhtml_Block_Widget_Grid_Column))
#8 app/code/community/BL/CustomGrid/Model/Grid/Rewriter/Eval.php(39) : eval()'d code(88): Mage_Adminhtml_Block_Widget_Grid->_setFilterValues(Array)
#9 app/code/core/Mage/Adminhtml/Block/Widget/Grid.php(534): BL_CustomGrid_Block_Rewrite_Mage_Adminhtml_Block_Catalog_Product_Grid->_setFilterValues(Array)
#10 app/code/core/Mage/Adminhtml/Block/Catalog/Product/Grid.php(125): Mage_Adminhtml_Block_Widget_Grid->_prepareCollection()
#11 app/code/community/BL/CustomGrid/Model/Grid/Rewriter/Eval.php(39) : eval()'d code(107): Mage_Adminhtml_Block_Catalog_Product_Grid->_prepareCollection()
#12 app/code/community/BL/CustomGrid/Model/Grid/Rewriter/Eval.php(39) : eval()'d code(434): BL_CustomGrid_Block_Rewrite_Mage_Adminhtml_Block_Catalog_Product_Grid->_prepareCollection()
#13 app/code/community/BL/CustomGrid/Model/Observer.php(500): BL_CustomGrid_Block_Rewrite_Mage_Adminhtml_Block_Catalog_Product_Grid->blcg_finishPrepareCollection()
#14 app/code/community/BL/CustomGrid/Model/Grid/Rewriter/Eval.php(39) : eval()'d code(104): BL_CustomGrid_Model_Observer->afterGridPrepareCollection(Object(BL_CustomGrid_Block_Rewrite_Mage_Adminhtml_Block_Catalog_Product_Grid))
#15 app/code/core/Mage/Adminhtml/Block/Widget/Grid.php(644): BL_CustomGrid_Block_Rewrite_Mage_Adminhtml_Block_Catalog_Product_Grid->_prepareCollection()
#16 app/code/core/Mage/Adminhtml/Block/Widget/Grid.php(651): Mage_Adminhtml_Block_Widget_Grid->_prepareGrid()
#17 app/code/core/Mage/Core/Block/Abstract.php(926): Mage_Adminhtml_Block_Widget_Grid->_beforeToHtml()
#18 app/code/core/Mage/Core/Block/Text/List.php(43): Mage_Core_Block_Abstract->toHtml()
#19 app/code/core/Mage/Core/Block/Abstract.php(927): Mage_Core_Block_Text_List->_toHtml()
#20 app/code/core/Mage/Core/Model/Layout.php(555): Mage_Core_Block_Abstract->toHtml()
#21 app/code/core/Mage/Core/Controller/Varien/Action.php(390): Mage_Core_Model_Layout->getOutput()
#22 app/code/core/Mage/Adminhtml/controllers/Catalog/ProductController.php(307): Mage_Core_Controller_Varien_Action->renderLayout()
#23 app/code/core/Mage/Core/Controller/Varien/Action.php(418): Mage_Adminhtml_Catalog_ProductController->gridAction()
#24 app/code/core/Mage/Core/Controller/Varien/Router/Standard.php(254): Mage_Core_Controller_Varien_Action->dispatch('grid')
#25 app/code/core/Mage/Core/Controller/Varien/Front.php(174): Mage_Core_Controller_Varien_Router_Standard->match(Object(Mage_Core_Controller_Request_Http))
#26 app/code/core/Mage/Core/Model/App.php(381): Mage_Core_Controller_Varien_Front->dispatch()
#27 app/Mage.php(745): Mage_Core_Model_App->run(Array)
#28 index.php(109): Mage::run('soh_nl', 'store')
#29 {main}
```

and the column is added here

```

            // add warehouse column to product grid
            $block->addColumnAfter('warehouse_id',
                array(
                    'header' => Mage::helper('safemage_multiinventory')->__('Warehouse'),
                    'type'  => 'options',
                    'sortable' => false,
                    'index' => 'warehouse_id',
                    'options' => $this->getActiveWarehousesArray(),
                    'filter_condition_callback' => array($this, 'filterWarehouseCondition'),
                    'width' => '120px'
                ),
                'sku'
            );

```

and used here as a filter

```

    public function filterWarehouseCondition($collection, $column)
    {
        if (!$value = $column->getFilter()->getValue()) {
            return;
        }

        $collection->addFilter('warehouse_tbl.warehouse_id', array('in' => $value));
    }

```

The problem is that the grid is using the index attribute name `'index' => 'warehouse_ids'` and not `'filter_condition_callback' => array($this, 'filterProductWarehouseCondition')` when filtering ... 

References:
https://magento.stackexchange.com/questions/294824/magento-1-9-4-3-custom-column-filter-not-working
https://magento.stackexchange.com/questions/292092/supee-11219-potential-issues/292307#292307
https://stackoverflow.com/questions/32222310/magento-invoice-grid-filter-condition-callback-not-working
and https://www.garybell.co.uk/fix-magento-invalid-attribute-error-supee-11219/

**Comment by seansan** [comment]
https://magento.stackexchange.com/questions/301420/magento-grid-filter-condition-callback-ignored-in-magento-ver-1-9-4-3

**Comment by sreichel** [comment]
See #887

**Comment by Flyingmana** [comment]
Reopening, because this sounds like a BC break, which we need to either prevent, or document cleanly enough

**Comment by sreichel** [comment]
@Flyingmana this issue/PR reverts a official patch that can be avoided/solved if you write your code correctly. Should be closed.

**Comment by sreichel** [comment]
Ref #1041

**Comment by colinmollenhour** [comment]
@Flyingmana I don't think any code was ever merged for this issue, and the PR #887 was actually a BC fix rather than a BC break.

I don't know the reasons for adding the extra clause in the first place but I am guessing it is to prevent executing code in arbitrary files although I don't see how it would be exploited without access to source code anyway.. Perhaps there is some extension that lets users add columns and define the callbacks. Anyway I think this issue is water under the bridge and the people that experienced a BC break in 1.9.3.x (from official patch, not OpenMage) have already fixed hte problem and moved on.
